### PR TITLE
Add alarms for malware detection

### PR DESCRIPTION
**Pull request by andrewleith**
# Summary | Résumé

This PR adds a warning and a critical alarm for when malware is detected.

- Warning occurs when 1 event of malware detection happens in a 1 minute period
- Critical occurs when 10 events of malware detection happens in a 1 minute period

### CODE REVIEW

File: aws/eks/cloudwatch_log.tf [hunk 1]
```diff
@@ -33,3 +33,15 @@ resource "aws_cloudwatch_log_metric_filter" "celery-error" {
     value     = "1"
   }
 }
+
+resource "aws_cloudwatch_log_metric_filter" "malware-detected" {
+  name           = "malware-detected"
+  pattern        = "?\"ERROR/Worker\" ?\"ERROR/ForkPoolWorker\" ?\"WorkerLostError\""
```
Comment (smcmurtry) [inline aws/eks/cloudwatch_log.tf:39]: This is the same pattern as the celery error above: https://github.com/cds-snc/notification-terraform/pull/659/files#diff-40738109cc9f7aa8ec6a2a1f292fa717335630a1697dc151af763b63573ab682R27

I think this pattern needs to match the malware detected log right? Or am I missing something?

Comment (andrewleith) [reply aws/eks/cloudwatch_log.tf:39]: Whoops, thank you, must have over-written my change here!

Comment (andrewleith) [reply aws/eks/cloudwatch_log.tf]: Wouldn't have gotten too far like that! 😅 Just pushed a commit to do this properly!

Comment (jzbahrai) [reply aws/eks/cloudwatch_log.tf]: Should we make the alarms less slack? Like warning - 10 files in a 5 minute period
critical - 20 files in a 5 minute period?

I am unsure about the 1 file in 1 minute warning. Though the 10 files in 1 minute can remain for critical

File: aws/eks/cloudwatch_log.tf [hunk 2]
```diff
@@ -33,3 +33,15 @@ resource "aws_cloudwatch_log_metric_filter" "celery-error" {
     value     = "1"
   }
 }
+
+resource "aws_cloudwatch_log_metric_filter" "malware-detected" {
+  name           = "malware-detected"
+  pattern        = "Malicious content detected"
```
Comment (smcmurtry) [inline aws/eks/cloudwatch_log.tf:39]:
```suggestion
  pattern        = "Malicious content detected! Download and attachment failed"
```

Could we target this to either just the celery log or just the dd-api log? I think this pattern will match on both logs and then we will be double counting. My suggested code just matches the log from Celery.

Comment (andrewleith) [reply aws/eks/cloudwatch_log.tf:39]: Sure! I wasn't sure how to target the logs (it just seemed to be looking at the entire EKS log group).  In my clickops testing in staging it didn't double count it, but I like your idea of making it more explicit.  I will make this update!

Comment (andrewleith) [reply aws/eks/cloudwatch_log.tf]: Done.

File: aws/eks/cloudwatch_log.tf (hunk 1 above)
Comment (andrewleith) [reply aws/eks/cloudwatch_log.tf]: Good question @jzbahrai. I think the idea here is that this is rare enough to warrant knowing about it even when there is just 1.  But maybe @smcmurtry you could give your thoughts?

**Review by jimleroyer — APPROVED** [review]
LGTM! 

Will we add alarms (could be just warnings) if we got timeout from the malware scanning? Steve is almost getting these changes in.

**Comment by andrewleith** [comment]
> LGTM!
> 
> Will we add alarms (could be just warnings) if we got timeout from the malware scanning? Steve is almost getting these changes in.

Certainly can.  Once that's done I can do another PR to add those.

**Pull request merged**